### PR TITLE
profile page created with profile info changing. Error with large images provided

### PR DESCRIPTION
**Pull request by Radon-Rodion**
pictures of large size can not be sent with post request (error 413: file is to large). Could not find param in webpack configs that defines max request size

### CODE REVIEW

File: src/api/serverOperations/findUserInfo.ts [hunk 1]
```diff
@@ -1,7 +1,28 @@
...
 }
 
-export default findUserInfo;
+export function findUserInfoByName(userName: string, usersList: Array<IUserInfo>): IUserInfo | undefined {
+  return usersList.find(((user) => user.userName === userName) ?? undefined);
+}
+
+export function findIndexById(id: number, usersList: Array<IUserInfo>): number {
+  // binary search with a specific
+  if (usersList.length === 0) return -1;
```
Comment (stnslvtvrn) [inline src/api/serverOperations/findUserInfo.ts]: if (!usersList.length) return -1;

File: src/elements/inputText/inputText.tsx [hunk 2]
```diff
@@ -11,10 +11,12 @@ interface IInputTextProps {
...
+  onChange: ((value: React.SetStateAction<string>) => void) | ((value: string) => void);
 }
 
-export function createChangeProcessor(setter: (value: React.SetStateAction<string>) => void) {
+export function createChangeProcessor(
+  setter: ((value: React.SetStateAction<string>) => void) | ((value: string) => void)
+) {
   const changeProcessor = (e: FormEvent) => {
     setter((e.target as HTMLInputElement).value);
   };
```
Comment (stnslvtvrn) [inline src/elements/inputText/inputText.tsx]: image

File: src/elements/pictureChoose/pictureChoose.tsx [hunk 3]
```diff
@@ -1,17 +1,37 @@
...
+    reader.onloadend = (event: ProgressEvent<FileReader>) => {
+      if (event?.target?.result != null) props.onChange(event?.target?.result as string);
+    };
+    if (files != null) reader.readAsDataURL(files[0]);
+  };
+
+  const pictureSrc = props.picture === "" ? noImg : props.picture;
+
+  return (
+    <div className={props.className}>
```
Comment (stnslvtvrn) [inline src/elements/pictureChoose/pictureChoose.tsx]: send instructions in skype message

File: src/pages/users/changePassword.tsx [hunk 4]
```diff
@@ -1,34 +1,29 @@
...
-  string,
-  React.Dispatch<React.SetStateAction<string>>,
-  string,
-  React.Dispatch<React.SetStateAction<string>>
-] = () => {
+const ChangePassword = (props: IChangePasswordProps) => {
   const [password, setPassword] = useState("");
   const [password2, setPassword2] = useState("");
 
-  return [password, setPassword, password2, setPassword2];
```
Comment (stnslvtvrn) [inline src/pages/users/changePassword.tsx]: newPassword/oldPassword

File: src/pages/users/signUp.tsx [hunk 5]
```diff
@@ -9,23 +9,10 @@ import { putUserInfo } from "@/api/clientRequests/postPutUserInfo";
...
-  React.Dispatch<React.SetStateAction<string>>,
-  string,
-  React.Dispatch<React.SetStateAction<string>>,
-  string,
-  React.Dispatch<React.SetStateAction<string>>
-] = () => {
+const SignUp = (props: ISignFormProps) => {
   const [login, setLogin] = useState("");
   const [password, setPassword] = useState("");
   const [password2, setPassword2] = useState("");
```
Comment (stnslvtvrn) [inline src/pages/users/signUp.tsx]: same with naming

**Review by stnslvtvrn — CHANGES_REQUESTED** [review]

File: src/api/serverOperations/findUserInfo.ts (hunk 1 above)
Comment (Radon-Rodion) [reply src/api/serverOperations/findUserInfo.ts]: redone as you adviced

File: src/elements/inputText/inputText.tsx (hunk 2 above)
Comment (Radon-Rodion) [reply src/elements/inputText/inputText.tsx]: made as you said

File: src/elements/pictureChoose/pictureChoose.tsx (hunk 3 above)
Comment (Radon-Rodion) [reply src/elements/pictureChoose/pictureChoose.tsx]: instructions followed. Photo is saved in localStorage

File: src/pages/users/signUp.tsx (hunk 5 above)
Comment (Radon-Rodion) [reply src/pages/users/signUp.tsx]: replaced with password and repeat password

File: src/pages/users/changePassword.tsx (hunk 4 above)
Comment (Radon-Rodion) [reply src/pages/users/changePassword.tsx]: replaced as you proposed

**Review by stnslvtvrn — APPROVED** [review]

**Pull request merged**